integration-cli/docker_cli_images_test.go

```diff
@@ -201,3 +201,109 @@ func (s *DockerSuite) TestImagesEnsureDanglingImageOnlyListedOnce(c *check.C) {
 	}
 
 }
+
+// See issue docker/docker#8926
+func (s *DockerSuite) TestImagesEnsureDanglingImagesExcludePullingImages(c *check.C) {
+	testRequires(c, Network)
+
+	cmd := exec.Command(dockerBinary, "images", "-q", "-f", "dangling=true")
+	initialOut, _, err := runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("listing images failed with errors: %s, %v", initialOut, err)
+	}
+	images := strings.Split(initialOut, "\n")
+	sort.Strings(images)
+	imagesStr := strings.Join(images, " ")
+
+	signalchan := make(chan error)
+	go func() {
+		defer close(signalchan)
+		pullCmd := exec.Command(dockerBinary, "pull", "hello-world")
+		if out, _, err := runCommandWithOutput(pullCmd); err != nil {
+			signalchan <- fmt.Errorf("Failed to pull %v: error %v, output %q", "hello-world", err, out)
+		}
+	}()
+
+	end := 0
+	for {
+		select {
+		case message, ok := <-signalchan:
```

Comment:
I think a cleaner implementation here would be to send nil to the channel above when there is no error instead of checking if the channel is closed or not.
Then it becomes:
```go
case err := <- signalChan:
    if err != nil {
        c.Fatal(err)
    }
```